libs/auth/src/cast_vote_record_hashes.ts

```diff
@@ -0,0 +1,335 @@
+import { Buffer } from 'buffer';
+import fs from 'fs/promises';
+import { sha256 } from 'js-sha256';
+import path from 'path';
+import { assert, groupBy } from '@votingworks/basics';
+import { Client } from '@votingworks/db';
+
+/**
+ * A representation of a file
+ */
+export interface File {
+  fileName: string;
+  fileContents: string | Buffer;
+}
+
+/**
+ * Computes a hash for a single cast vote record, incorporating the cast vote record's directory
+ * name, file names, and file contents. The directory summary that we hash specifically mirrors the
+ * output of:
+ * find <directoryName> -type f | sort | xargs sha256sum
+ */
+export function computeSingleCastVoteRecordHash({
+  directoryName,
+  files,
+}: {
+  directoryName: string;
+  files: File[];
+}): string {
+  const filesSorted = [...files].sort((file1, file2) =>
+    file1.fileName.localeCompare(file2.fileName)
+  );
+  const fileHashes: string[] = [];
+  for (const { fileName, fileContents } of filesSorted) {
+    const filePath = path.join(directoryName, fileName);
+    // Be extra cautious and prevent spoofing the directory summary by using directory/file names
+    // with newlines
+    assert(!filePath.includes('\n'));
```

Comment:
I actually don't think so because `\r` and `\n` are different values with different hashes. So for example, the following will have different hashes:

```
abcd  file1.json\n
efgh  file2.json\n
```

```
abcd  file1.json\r
efgh  file2.json\n
```

There isn't a way for someone to spoof a directory summary separated by `\n` values using `\r` values.

But a good sanity check!

Reply:
And thank you for the other change! Merged it into this branch :100: